### PR TITLE
dotnet ecosystem: various fixes

### PR DESCRIPTION
**Pull request by IvarWithoutBones**
###### Description of changes
This fixes a few annoying bugs in `buildDotnetModule` and some of the functions it uses. Most notably flags like `dotnetFlags` and `makeWrapperArgs` we not properly interpreted from the child derivation, causing some issues if they contained spaces. As a workaround some derivations set those flags through bash, but that's obviously not great. I'll create a PR fixing those after this gets merged :+1: 

For more information see the commit descriptions. 

cc @SuperSandro2000 @Mic92 @jonringer @prusnak 

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [X] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by jonringer** [comment]
> By the way, do any of you happen to know if I could add `pkgs/build-support/dotnet` to CODEOWNERS? I'd like to get notified and review the changes when someone touches it.

This works if you have commit rights, otherwise it won't

**Comment by jonringer** [comment]
looks like some issues with the v3:

```
error: builder for '/nix/store/slk3sjikvjjdsqd5c87mxf4c5ch85mwj-ppy.osu.Game.Resources-2022.407.0.nupkg.drv' failed with exit code 1;
       last 7 log lines:
       >
       > trying https://www.nuget.org/api/v3/package/ppy.osu.Game.Resources/2022.407.0
       >   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
       >                                  Dload  Upload   Total   Spent    Left  Speed
       >   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
       > curl: (22) The requested URL returned error: 404
       > error: cannot download ppy.osu.Game.Resources-2022.407.0.nupkg from any mirror

**Comment by IvarWithoutBones** [comment]
> This works if you have commit rights, otherwise it won't

I do, I'll make a PR for that later :+1: 

> looks like some issues with the v3:

Ah, looks like that only worked locally because I had all nuget deps cached already. I'll drop that commit for now.

File: pkgs/build-support/dotnet/build-dotnet-module/default.nix [hunk 1]
```diff
@@ -74,13 +74,16 @@ let
...
+    then linkFarmFromDrvs "${name}-project-references" projectReferences
+    else null;
+
+  _nugetDeps = mkNugetDeps { inherit name; nugetDeps = import nugetDeps; };
 
   nuget-source = mkNugetSource {
-    name = "${args.pname}-nuget-source";
-    description = "A Nuget source with the dependencies for ${args.pname}";
-    deps = [ _nugetDeps _localDeps ];
+    name = "${name}-nuget-source";
```
Comment (SuperSandro2000) [inline pkgs/build-support/dotnet/build-dotnet-module/default.nix:84]: Does that invalidate with version updates?

Comment (SuperSandro2000) [inline pkgs/build-support/dotnet/build-dotnet-module/default.nix:84]: Does that invalidate with version updates?

Comment (jonringer) [reply pkgs/build-support/dotnet/build-dotnet-module/default.nix:84]: It should. Derivation name influences the sha. So new versions will force a new realization.

**Comment by IvarWithoutBones** [comment]
I've pushed some more fixes for `mkNugetSource`'s metadata generation :+1:

**Review by kira64xyz — APPROVED** [review]

**Comment by IvarWithoutBones** [comment]
I have cherry-picked the `nuget-to-nix` commit from #170300 as well, with the idea that I'll work on lockfile generation and the passthru update script next. I figured it'd be best to try to get those changes in before #170300 so that it can stabalise and we don't needlessly invalidate any hashes.

**Review by jonringer — APPROVED** [review]
LGTM
failure is from gcc 11 bump

```
https://github.com/NixOS/nixpkgs/pull/170300

1 package blacklisted:
tests.writers

1 package failed to build:
netcoredbg

17 packages built:
ArchiSteamFarm alttpr-opentracker baget btcpayserver depotdownloader discordchatexporter-cli formula inklecate jackett nbxplorer nuget-to-nix opentabletdriver osu-lazer pinta python-language-server ryujinx wasabibackend
```

**Pull request merged**